src/tools/registry/gen_inc.c

```diff
@@ -83,6 +109,122 @@ int write_field_pointer_arrays(FILE* fd){/*{{{*/
 	return 0;
 }/*}}}*/
 
+// Checks for a string in a list of strings.
+// Returns the index of the string if it does exist in the array,
+// and -1 if it does not appear in the array.
+int find_string_in_array(char *input_string, const char *array[], size_t rows){
+	size_t i;
+	for (i = 0; i < rows; i++ ){
+		if (strcmp(input_string, array[i]) == 0){
+			return i;
+		}
+	}
+	return -1;
+}
+
+// Helper function to change attribute names in accordance with
+// "attrs_to_modify" within the "add_attribute_if_not_ignored" function
+const char * modify_attr(const char *attr, const char *array[][2], size_t rows) {
+	size_t i;
+	for (i = 0; i < rows; i++) {
+		if (strcmp(attr, array[i][0]) == 0) {
+			return array[i][1];
+		}
+	}
+	return attr;
+}
+
+
+// Doubles single quotes in stringIn, and places the results in the buffer
+// stringOut. stringOut should be large enough to store (len(stringIn) * 2 + 1)
+// characters. Returns 1 if the buffer is too small for the result. 
+int escape_quotes(const char * stringIn, char * result, size_t bufferSize){
+	size_t resultIndex = 0;
+	
+	for (size_t i = 0; i < strlen(stringIn); i++) {
+		if ( stringIn[i] == '\'' ) {
+			if (resultIndex + 1 > bufferSize) return 1;
+			result[resultIndex++] = '\'';
+		}
+		if (resultIndex + 1 > bufferSize) return 1;
+		result[resultIndex++] = stringIn[i];
+	}
+
+	if (resultIndex + 1 > bufferSize) return 1;
+	result[resultIndex++] = '\0';
+	
+	return 0;
+}
+
+void add_attribute_if_not_ignored(FILE *fd, char *index, char *att_name, char *pointer_name_arr, char *att_value){	
+	char *format_string;
+	
+	// Allocate buffers for escaping apostrophes, 
+	size_t value_buffer_size = 2 * strlen(att_value) + 1;
+	size_t name_buffer_size = 2 * strlen(att_name) + 1;
+	char *escaped_value = (char*)malloc(value_buffer_size);
+	char *escaped_name = (char*)malloc(name_buffer_size);
+	int err;
+	
+	// Confirm that memory was allocated correctly
+	if (escaped_value == NULL) {
+		fprintf(stderr,
+ 		     	"ERROR: Failed to allocate memory while escaping quotes for att_value %s of att %s\n",
+ 			   	att_value,
+ 			   	att_name);
+		return;
+	} else if (escaped_name == NULL) {
+		fprintf(stderr,
+ 			   	"ERROR: Failed to allocate memory while escaping quotes for att_name of att %s\n",
+				att_name);
+		return;
+	}
+
+
+	// Return early if we want to ignore the attribute
+	if (find_string_in_array(att_name, ATTRS_TO_IGNORE, NUM_IGNORED_ATTRS) >= 0){
+		return;
+	}
+
+	// check if the attribute is numeric
+	if (find_string_in_array(att_name, NUMERIC_ATTRS, NUM_NUMERIC_ATTRS) >= 0){
+		format_string = "      call mpas_add_att(%s %% attLists(%s) %% attList, '%s', %s)\n";
+	} 
+	// If it isn't numeric, make sure to wrap  att_value in quotes
+	else {
+ 		format_string = "      call mpas_add_att(%s %% attLists(%s) %% attList, '%s', '%s')\n";
+	}
+	
+	// Escape the quotes
+	err = escape_quotes(att_value, escaped_value, value_buffer_size);
+	if ( err == 1 ){
```

Comment:
We could save having to declare the local variable `err` if the conditional just used the call to `escape_quotes` on the left of the comparison (and similarly for the if-test on line 211).